include/dem/ray_tracing.h

```diff
@@ -0,0 +1,198 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+#ifndef lethe_ray_tracing_h
+#define lethe_ray_tracing_h
+
+#include <core/dem_properties.h>
+
+#include <dem/data_containers.h>
+#include <dem/dem_contact_manager.h>
+#include <dem/insertion.h>
+#include <dem/load_balancing.h>
+#include <dem/ray_tracing_solver_parameters.h>
+
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/timer.h>
+
+template <int dim>
+class RayTracingSolver
+{
+public:
+  RayTracingSolver(RayTracingSolverParameters<dim> parameters,
+                   DEMSolverParameters<dim>        dem_parameters);
+
+  /**
+   * @brief Calls all the necessary functions to set parameters, solve the intersection
+   * points between the photons and the particles, and finish the simulation.
+   */
+  void
+  solve();
+
+private:
+  /**
+   * @brief Set the parameters for the ray tracing.
+   */
+  void
+  setup_parameters();
+
+  /**
+   * @brief Set the insertion method.
```

Comment:
Can you describe rapidly what insertion methods are supported? Do you insert particles like in DEM or do you restart from a DEM simulation? It would important to describe this rapidly here